src/main/java/io/github/jhipster/util/ReflectUtils.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016-2017 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jhipster.util;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * This utility is intended to be used with optional dependencies that are only used in certain
+ * Spring profiles, such as with Liquibase and Swagger, and potentially stripped entirely from
+ * the classpath. Invoking methods via reflection avoids compile-time errors in those cases.
+ */
+public interface ReflectUtils {
```

Comment:
Btw I think Spring has a built in reflection utils not sure if it will match the requirements but worth a look

Reply:
I looked at it briefly but it seemed not to do class loading (and we want to only know about class names) and a bit awkward in handling exceptions. Just my opinion after a cursory glance, mind you.

Reply:
I think it can be used for invoke(...) functions